### PR TITLE
Api 7888 create production access form

### PR DESCRIPTION
**Pull request by keithosayande**
### Description
This PR handles the creation of the UI for the Production Access From. The design for the form can be found [here](https://www.figma.com/file/9sOD2GDAsyNTayvOqKlcJV/API-3854-Consumer?node-id=1315%3A479). The content and logic can be found [here](https://docs.google.com/document/d/1cMGSHFXd4ftVGQpJTqHvQgkbh4n63sVZoCF15fhbSvs/edit). This PR also contains the addition of Open API tags.

### Process

- [ ] Tests added or updated for change
- [ ] Docs updated
  - [ ] If you have made structural changes to the site, the [Developer Portal Content Types](https://community.max.gov/x/qUwOg) docs have been updated
  - [ ] If you have made/confirmed any decisions about user interactions and accessibility, the [Accessible Component Design](https://community.max.gov/x/xS8mgg) docs have been updated
- [ ] Accessibility concerns have been considered and addressed
  
  - [ ] If you've made any UI changes, you've run an axe check using the axe browser extension that reported no new issues

### CODE REVIEW

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    email: string;
+  };
+  organization: string;
+  phoneNumber: string;
+  applicationName?: string;
+  statusUpdateEmails: string[];
+  valueProvided: string;
+  businessModel?: string;
+  hasMonetized: string;
+  monetizationExplination?: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:49]: `hasMonetized`: looks like this is a boolean on the backend, and also named `monitizedVeteranInformation`?

`monetizationExplination`: typo or spelling error in "explanation" (a instead of i). also, to cap that off, we've got a mismatch here too because looks like "monetization" is spelled with an i instead of an e on the backend? so we should probably get those all lined up, spelled correctly or not

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+  isVetFacing: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:50]: appears to be named `veteranFacing` and typed as a boolean on the backend

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+  website?: string;
+  signUpLink?: string[];
+  supportLink?: string[];
+  platforms?: string;
+  appDescription?: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:55]: okay, i'm confused what's going on here. in the backend we have `appDescription`, which is required, and `veteranFacingDescription`, which is not and appears to have the validations listed at that field in the doc. i can't figure out what `appDescription` in the backend refers to.

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+  vasiSystemName?: string;
+  credentialStorage: string;
+  storePIIOrPHI: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:58]: does this get asked for open data APIs? i know it's marked required on the backend

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+  piiStorageMethod?: string;
+  multipleReqSafeguards?: string;
+  breachManagementProcess?: string;
+  vulnerabilityManagement?: string;
+  exposesToThirdParties: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:63]: this is `exposeVeteranInformationToThirdParties` on the backend, i believe

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+  thirdPartyInfoDescription?: string;
+  scopesAccessRequested?: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:65]: appears to be an array on the backend, although this looks right for the UI

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+  distributingAPIKeysToCustomers?: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:66]: marked required on the backend but it appears optional due to the fact it's only relevant to Benefits Intake

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+  namingConvention?: string;
+  centralizedBackendLog?: string;
+  listedOnMyHealthApplication?: string;
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:69]: looks like a boolean on the backend

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+  policyDocuments?: string[];
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:70]: also appears required on the backend for the same reason

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+
+      calculateSteps(values);
+      setActiveStep(activeStep + 1);
+      actions.setTouched({});
+      actions.setSubmitting(false);
+    }
+  };
+  return (
+    <div className={classNames('vads-l-grid-container', 'vads-u-padding--4')}>
+      <PageHeader header={headerText} />
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:185]: style comment, optional: think this header text is small enough that it could be included inline

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+          <Formik
+            initialValues={initialValues}
+            onSubmit={handleSubmit}
+            validationSchema={currentValidationSchema}
+            validateOnBlur={false}
+            validateOnChange={false}
+          >
+            {({ submitForm }): React.ReactNode => {
+              const handleSubmitButtonClick = async (): Promise<void> => {
+                await submitForm();
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:197]: i think we're going to want to take out this explicit `submitForm` call. Quokka just resolved a bug in the Apply form where it was submitting twice (which i'm guessing you might have looked at when writing this form, maybe? commit [here](https://github.com/department-of-veterans-affairs/developer-portal/commit/bdebe3bfd978876efb4eb0267c688e141a779d03)). @christineyost feel free to add anything about that if there's anything else to understand about that setup

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+                    errorElements[0].focus();
+                  }
+                }, 0);
+              };
+              return (
+                <Form>
+                  {activeStep === 0 ? (
+                    <>
+                      <SegmentedProgressBar current={1} total={4} />
+                      <h4>Step 1: Verification</h4>
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:212]: this is going to have to be an h2, even if it's styled to look like an h4. it's important to keep a semantic heading order that reflects the document hierarchy because screen reader users use the heading level to get an understanding of page structure

File: src/containers/consumerOnboarding/productionAccessFormSteps/BasicInformation.tsx [hunk 13]
```diff
@@ -0,0 +1,243 @@
+import React, { FC } from 'react';
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/productionAccessFormSteps/BasicInformation.tsx:1]: might suggest moving this file into `consumerOnboarding/components/productionAccessForm[or whatever]/BasicInformation.tsx`. see `consumerOnboarding/components/sandbox` and `publishing/components` for examples of that pattern. i'm okay with any structure though.

File: src/containers/consumerOnboarding/productionAccessFormSteps/SelectedApis.tsx [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+    {apiCheckboxes.map(api => (
+      <Flag name={[FLAG_HOSTED_APIS, api.urlFragment]} key={api.urlFragment}>
+        <CheckboxRadioField
+          type="checkbox"
+          name="apis"
+          label={
+            <>
+              <span>{api.name}</span>
+              <span className="vads-u-display--inline-block vads-u-margin-left--1">
+                {api.vaInternalOnly || api.trustedPartnerOnly || api.openData ? (
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/productionAccessFormSteps/SelectedApis.tsx:25]: couple thoughts here:

1. you can leave off the undefined by using `&&` instead of a ternary if you prefer
2. does `OnlyTags` render nothing if none of the values are true? if so can leave off the conditional

File: src/containers/consumerOnboarding/productionAccessFormSteps/SelectedApis.tsx [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+  const { errors } = useFormikContext();
+  const checkboxName = 'apis';
+  const shouldDisplayErrors = !!errors[checkboxName];
+  const containerClass = shouldDisplayErrors ? 'usa-input-error' : '';
+  const labelClass = shouldDisplayErrors ? 'usa-input-error-label' : '';
+  const validationClass = shouldDisplayErrors ? 'usa-input-error-message' : '';
+  const errorMessagePaddingClass = shouldDisplayErrors ? 'vads-u-padding-x--1p5' : '';
+  const selectAPIClass = shouldDisplayErrors
+    ? 'vads-u-font-weight--bold'
+    : 'vads-u-font-weight--normal';
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/productionAccessFormSteps/SelectedApis.tsx:53]: `classNames` is good at this kind of logic if you prefer (especially for ones where the default is no class). like this

```
classNames({
  'usa-input-error': shouldDisplayErrors,
  ...
})
```

**Review by mjlumetta — COMMENTED** [review]
didn't review everything but here are some comments for monday

File: src/containers/consumerOnboarding/validateProductionAccessForm.ts [hunk 16]
```diff
@@ -0,0 +1,43 @@
+import { FormikErrors } from 'formik';
+import { includesOAuthAPI } from '../../apiDefs/query';
+import { APPLY_FIELDS_TO_URL_FRAGMENTS } from '../../types/constants';
+import { validateYesOrNo, validatePresence } from '../../utils/validators';
+import { Values } from './ProductionAccess';
+
+export const anyOAuthApisSelected = (apis: string[]): boolean => {
+  const apiIdsByField = apis.flatMap(formField => APPLY_FIELDS_TO_URL_FRAGMENTS[formField]);
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/validateProductionAccessForm.ts:8]: can this just be `map`, or am i missing something?

File: src/containers/consumerOnboarding/validateProductionAccessForm.ts [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+const anyApiSelected = ({ apis }: Values): boolean => apis.length > 0;
+
+export const validateProductionAccessForm = (values: Values): FormikErrors<Values> => {
+  const errors: FormikErrors<Values> = {
+    is508Compliant: validateYesOrNo(values.is508Compliant),
+    isUSBasedCompany: validateYesOrNo(values.isUSBasedCompany),
+    primaryContact: {
+      email: validatePresence('email', values.primaryContact.email),
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/validateProductionAccessForm.ts:19]: `validateEmail`?

also does the `secondaryContact` need to be validated as well?

File: src/containers/consumerOnboarding/validationSchema.ts [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+import * as Yup from 'yup';
+import { includesInternalOnlyAPI, includesOAuthAPI } from '../../apiDefs/query';
+
+const phoneRegex = /^(?:\+?1[-.●]?)?\(?([0-9]{3})\)?[-.●]?([0-9]{3})[-.●]?([0-9]{4})$/;
+
+const validationSchema = [
+  Yup.object().shape({
+    apis: Yup.array().of(Yup.string()).min(1).required('Choose at least one API.'),
+    is508Compliant: Yup.string()
+      .matches(/^(?:yes|no)$/)
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/validationSchema.ts:12]: can this be `oneOf` instead of `matches` since it's two discrete values?

File: src/containers/releaseNotes/CategoryReleaseNotes.tsx [hunk 19]
```diff
@@ -46,7 +47,7 @@ const ReleaseNotesCardLinks: React.FunctionComponent<ReleaseNotesCardLinksProps>
               name={name}
               subhead={
                 vaInternalOnly || trustedPartnerOnly ? (
-                  <OnlyTags {...{ trustedPartnerOnly, vaInternalOnly }} />
+                  <ApiTags {...{ openData, trustedPartnerOnly, vaInternalOnly }} />
```
Comment (mjlumetta) [inline src/containers/releaseNotes/CategoryReleaseNotes.tsx:50]: it's important to note that this change _will_ get deployed when this branch is merged (and another deploy goes out). @ala-yna @wwanamaker @brentschneider

File: src/utils/validators.ts [hunk 20]
```diff
@@ -33,3 +35,11 @@ export const validateOAuthRedirectURI = (value: string): string | undefined => {
 
   return undefined;
 };
+
+export const validateYesOrNo = (value: string): string | undefined => {
+  if (!YES_OR_NO_PATTERN.test(value)) {
```
Comment (mjlumetta) [inline src/utils/validators.ts:40]: any opposition to `['yes', 'no'].includes(value)` since it can be captured in discrete values? (this is fine, but don't necessarily have to make a regex in this case)

File: src/types/constants/paths.ts [hunk 21]
```diff
@@ -13,7 +13,6 @@ export const CONSUMER_ROUTER_PATHS: string[] = [
   CONSUMER_PROD_PATH,
   CONSUMER_DEMO_PATH,
   CONSUMER_APIS_PATH,
-  CONSUMER_APPLICATION_PATH,
```
Comment (mjlumetta) [inline src/types/constants/paths.ts:16]: why was this removed?

File: src/containers/consumerOnboarding/validateProductionAccessForm.ts [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+import { Values } from './ProductionAccess';
+
+export const anyOAuthApisSelected = (apis: string[]): boolean => {
+  const apiIdsByField = apis.flatMap(formField => APPLY_FIELDS_TO_URL_FRAGMENTS[formField]);
+  return includesOAuthAPI(apiIdsByField);
+};
+
+const anyApiSelected = ({ apis }: Values): boolean => apis.length > 0;
+
+export const validateProductionAccessForm = (values: Values): FormikErrors<Values> => {
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/validateProductionAccessForm.ts:14]: is this not being used in favor of the `validationSchema` prop? if so can it be deleted?

**Review by mjlumetta — COMMENTED** [review]
more comments, mostly about style/cleanliness so a lot can be optional. still need to do some testing

File: src/containers/consumerOnboarding/components/productionAccessForm/Verification.tsx [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+      errorClassName="vads-u-margin-left--2"
+      name="is508Compliant"
+      required
+    >
+      <CheckboxRadioField type="radio" label="Yes" name="is508Compliant" value="yes" required />
+      <CheckboxRadioField type="radio" label="No" name="is508Compliant" value="no" required />
+    </FieldSet>
+
+    <SelectedAPIs />
+    <CheckboxRadioField
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/components/productionAccessForm/Verification.tsx:41]: the `CheckboxRadioField` component has no built in error display, so we either need to build that in or do a custom error display here (go custom if you have any doubts). but yeah, it doesn't let you go forward, but it doesn't give you any feedback either

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 24]
```diff
@@ -0,0 +1,307 @@
...
+    }
+  };
+  const handleSubmit = (values: Values, actions: FormikHelpers<Values>): void => {
+    setTimeout(() => {
+      const errorElements = document.querySelectorAll<HTMLElement>('[aria-invalid=true]');
+
+      if (errorElements.length > 0) {
+        errorElements[0].focus();
+      }
+    }, 0);
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/ProductionAccess.tsx:169]: i'm not sure this is working... might have to go back and double-check + debug, but yeah

File: src/containers/consumerOnboarding/components/productionAccessForm/BasicInformation.tsx [hunk 25]
```diff
@@ -0,0 +1,244 @@
...
+        className="vads-u-margin-top--4 medium-screen:vads-l-col--10"
+      />
+
+      <TextField
+        label="Email"
+        name="primaryContact.email"
+        type="email"
+        required
+        className="vads-u-margin-top--4 medium-screen:vads-l-col--10"
+      />
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/components/productionAccessForm/BasicInformation.tsx:42]: couple things here, both of which I believe are generalizable to other fields:

1. the message is, i assume, not the one we want. so there's a Yup thing to figure out there.
2. the red error displays aren't present. i am guessing this is a result of [this line](https://github.com/department-of-veterans-affairs/developer-portal/pull/814/files#diff-9a6083f636e285ed29d334fcd7baee3963ef2e2ef1d45e01c27a96d930fb18beR187) in `ProductionAccess` where the touched fields are cleared; if you look at line 39 of `TextField`, you'll see that errors only show for touched fields. also need to debug further.

Screen Shot 2021-08-04 at 6 02 13 PM

Comment (mjlumetta) [reply src/containers/consumerOnboarding/components/productionAccessForm/BasicInformation.tsx:42]: although sometimes the red highlighting works?? confused 😕  (specifically for phone number)

File: src/containers/consumerOnboarding/validationSchema.ts [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+      .required('Select yes or no.'),
+    monetizationExplination: Yup.string().when('hasMonetized', {
+      is: (value: string) => value === 'yes',
+      otherwise: Yup.string(),
+      then: Yup.string().required('Enter an explanation.'),
+    }),
+    organization: Yup.string().required('Enter the company or organization name.'),
+    phoneNumber: Yup.string()
+      .matches(phoneRegex, {
+        message: 'Enter a company phone number.',
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/validationSchema.ts:47]: this error message should probably say to enter a valid number specifically (and possibly offer formatting options) cc @brentschneider

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx [hunk 27]
```diff
@@ -0,0 +1,164 @@
...
+        {({ remove, push }): ReactNode => (
+          <div className="vads-u-padding-y--1p5">
+            {data.map((value, index) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <div key={index}>
+                <TextEntry
+                  name={name}
+                  index={index}
+                  label={label}
+                  onClick={(): void => remove(index)}
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:134]: couple things here: probably don't want to make it possible to remove the last one, and if this is specifically to handle the remove event (not a general click event), i would give it a name to reflect that specifically

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx [hunk 28]
```diff
@@ -0,0 +1,164 @@
...
+    <div
+      className={classNames(containerClass, 'vads-u-background-color--gray-lightest', className)}
+    >
+      {description}
+      <span
+        id="api-checkbox-error"
+        className={classNames(validationClass, errorMessagePaddingClass)}
+        role="alert"
+      >
+        <ErrorMessage name={name} />
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:122]: i do not know how this happened, but it did 😮 

Screen Shot 2021-08-04 at 6 20 46 PM

File: src/containers/consumerOnboarding/components/productionAccessForm/PolicyGovernance.tsx [hunk 29]
```diff
@@ -0,0 +1,27 @@
+import React, { FC } from 'react';
+import ListOfTextEntries from './ListOfTextEntries';
+
+const PolicyGovernance: FC = () => (
+  <>
+    <h3>Terms of service and privacy policies</h3>
+    <ListOfTextEntries
```
Comment (mjlumetta) [inline src/containers/consumerOnboarding/components/productionAccessForm/PolicyGovernance.tsx:7]: couple of edge conditions here:

1. how to validate empty elements in the `ListOfTextEntries`
2. error message showing on individual text field (probably don't want)

Screen Shot 2021-08-04 at 6 24 42 PM

**Review by mjlumetta — COMMENTED** [review]
okay, bunch of testing/functional comments here. i did not test every path, and i did not do screen reader testing, so those are to be done. i assume both can come after this PR is merged since there will be QA after merging to master, but want to note that. these are some of the bigger or more immediately clear issues that i could flag. again, don't need to QA perfectly before merge since we will swarm on it after, but the more we can find the more we can fix in this PR or plan for as a separate story.

File: src/containers/releaseNotes/CategoryReleaseNotes.tsx (hunk 19 above)
Comment (brentschneider) [reply src/containers/releaseNotes/CategoryReleaseNotes.tsx:50]: If I understand correctly, does that mean the tag "Open Data" will go live? I'm okay with that.

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 30]
```diff
@@ -0,0 +1,311 @@
...
+                      <button
+                        type="submit"
+                        className="usa-button-primary va-button-primary vads-u-width--auto"
+                        onClick={handleSubmitButtonClick}
+                      >
+                        Submit your application
+                      </button>
+                    ) : (
+                      <button
+                        type="submit"
```
[the remark this answers is not in the document]
Comment (keithosayande) [reply src/containers/consumerOnboarding/ProductionAccess.tsx]: Needs to be `type="submit"` for the validation to run.

**Review by johanna-adhoc — COMMENTED** [review]
Issues that I have encounter while using the form:
1- the progress bar is focusable even though the element doesn't do any action for the user. @brentschneider is this on purpose?
2- The error messages are not clearing on blur or state change.
3- I can't move forward past step 2 as no phone number is accepted as valid
4- On step 2 I am not filling out any of the items on the top section (name, email, etc) and no error message is displayed.

File: src/containers/releaseNotes/CategoryReleaseNotes.tsx (hunk 19 above)
Comment (johanna-adhoc) [reply src/containers/releaseNotes/CategoryReleaseNotes.tsx:50]: We will link the new Open Data tag with the consumer docs flag so it will only show visible once the new feature goes live

File: src/utils/validators.ts (hunk 20 above)
Comment (johanna-adhoc) [reply src/utils/validators.ts:40]: This code has been removed

File: src/containers/consumerOnboarding/validationSchema.ts (hunk 26 above)
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/validationSchema.ts:47]: text changed to Enter a valid company phone number

File: src/containers/consumerOnboarding/components/productionAccessForm/PolicyGovernance.tsx (hunk 29 above)
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/PolicyGovernance.tsx]: This will be address with new ticket API-9122

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx (hunk 28 above)
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:123]: This will be address with new ticket API-9122

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx (hunk 27 above)
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:135]: This will be address with new ticket API-9122

File: src/containers/consumerOnboarding/components/productionAccessForm/BasicInformation.tsx (hunk 25 above)
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/BasicInformation.tsx:48]: this has been addressed

File: src/containers/consumerOnboarding/components/productionAccessForm/Verification.tsx (hunk 23 above)
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/Verification.tsx:41]: This will be addressed by a new story

File: src/containers/consumerOnboarding/validationSchema.ts [hunk 31]
```diff
@@ -0,0 +1,175 @@
+/* eslint-disable newline-per-chained-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import * as Yup from 'yup';
+import { includesInternalOnlyAPI, includesOAuthAPI } from '../../apiDefs/query';
+
+const phoneRegex = /^(?:\+?1[-.●]?)?\(?([0-9]{3})\)?[-.●]?([0-9]{3})[-.●]?([0-9]{4})$/;
+
+const validationSchema = [
+  Yup.object().shape({
+    apis: Yup.array().of(Yup.string()).min(1).required('Choose at least one API.'),
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/validationSchema.ts:10]: This has been resolved

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx [hunk 32]
```diff
@@ -0,0 +1,164 @@
...
+  const containerClass = shouldDisplayErrors ? 'usa-input-error' : '';
+  const validationClass = shouldDisplayErrors ? 'usa-input-error-message' : '';
+  const errorMessagePaddingClass = shouldDisplayErrors ? 'vads-u-padding-x--1p5' : '';
+  return (
+    <div
+      className={classNames(containerClass, 'vads-u-background-color--gray-lightest', className)}
+    >
+      {description}
+      <span
+        id="api-checkbox-error"
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:119]: This id will be removed

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx [hunk 33]
```diff
@@ -0,0 +1,164 @@
...
+            }}
+            className={classNames(
+              'usa-button-secondary',
+              'vads-u-margin-bottom--0',
+              'vads-u-margin-left--neg9',
+              'vads-u-margin-right--0',
+              'vads-u-margin-top--0',
+            )}
+          >
+            Edit
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:82]: This will get addressed with API-9122

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx [hunk 34]
```diff
@@ -0,0 +1,164 @@
...
+  label?: string;
+  onClick: (event: MouseEvent) => void;
+}
+
+export interface ListOfTextEntriesProps {
+  description: ReactNode;
+  className?: string;
+  name: string;
+  buttonText: string;
+  label?: string;
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:20]: This will be address by API-9122

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 35]
```diff
@@ -0,0 +1,311 @@
...
+  const [steps, setSteps] = useState(possibleSteps);
+  const currentValidationSchema = validationSchema[activeStep];
+  const isLastStep = activeStep === steps.length - 1;
+
+  const { modalVisible: modal1Visible, setModalVisible: setModal1Visible } = useModalController();
+  const { modalVisible: modal2Visible, setModalVisible: setModal2Visible } = useModalController();
+  const { modalVisible: modal3Visible, setModalVisible: setModal3Visible } = useModalController();
+  const history = useHistory();
+
+  const calculateSteps = (values: Values): void => {
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/ProductionAccess.tsx:131]: No code change is needed so I will resolve

File: src/containers/consumerOnboarding/productionAccessFormSteps/ListOfTextEntries.tsx [hunk 36]
```diff
@@ -0,0 +1,195 @@
...
+  const containerClass = shouldDisplayErrors ? 'usa-input-error' : '';
+  const validationClass = shouldDisplayErrors ? 'usa-input-error-message' : '';
+  const errorMessagePaddingClass = shouldDisplayErrors ? 'vads-u-padding-x--1p5' : '';
+  return (
+    <div
+      className={classNames(containerClass, 'vads-u-background-color--gray-lightest', className)}
+    >
+      {description}
+      <span
+        id="api-checkbox-error"
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/productionAccessFormSteps/ListOfTextEntries.tsx]: This was fixed

File: src/containers/consumerOnboarding/validationSchema.ts (hunk 18 above)
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/validationSchema.ts:12]: this has been addressed

File: src/containers/consumerOnboarding/productionAccessFormSteps/ListOfTextEntries.tsx [hunk 37]
```diff
@@ -0,0 +1,195 @@
...
+};
+
+const ListOfTextEntries: FC<ListOfTextEntriesProps> = ({
+  description,
+  className,
+  name,
+  buttonText,
+  label,
+}) => {
+  const { values, errors } = useFormikContext<Values>();
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/productionAccessFormSteps/ListOfTextEntries.tsx]: This will get addressed by API-9122

File: src/containers/consumerOnboarding/productionAccessFormSteps/ListOfTextEntries.tsx [hunk 38]
```diff
@@ -0,0 +1,195 @@
...
+        role="alert"
+      >
+        <ErrorMessage name={name} />
+      </span>
+      <FieldArray name={name}>
+        {({ remove, push }): ReactNode => (
+          <div className="vads-u-padding-y--1p5">
+            {data.map((value, index) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <div key={index}>
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/productionAccessFormSteps/ListOfTextEntries.tsx]: resolving

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 39]
```diff
@@ -0,0 +1,307 @@
...
+        return;
+      }
+
+      if (values.is508Compliant === 'no') {
+        setModal3Visible(true);
+        return;
+      }
+
+      calculateSteps(values);
+      setActiveStep(activeStep + 1);
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/ProductionAccess.tsx:186]: this has already been fixed

File: src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx [hunk 40]
```diff
@@ -0,0 +1,164 @@
...
+        className={fieldClass}
+        customFieldClass={classNames('vads-u-margin-right--0')}
+      >
+        {disabled && (
+          <button
+            type="button"
+            name="edit"
+            onClick={(): void => {
+              setDisabled(false);
+              setIsEditing(true);
```
[the remark this answers is not in the document]
Comment (johanna-adhoc) [reply src/containers/consumerOnboarding/components/productionAccessForm/ListOfTextEntries.tsx:72]: Will be addressed with accessibility story

**Review by johanna-adhoc — APPROVED** [review]
LGTM

**Review by johanna-adhoc — APPROVED** [review]
